File: py-brainwords/brainwords.py
import math

def generateBFCode(text):
    
    result = ""
    lastchar = '\0'
    factor = 0
    remain = 0
    direction = '+'

    for c in text:

        if ord(c) > ord(lastchar):
            factor = math.floor(math.sqrt(ord(c) - ord(lastchar)))
            remain = (ord(c) - ord(lastchar)) - factor * factor
            direction = '+'
        else:
            factor = math.floor(math.sqrt(ord(lastchar) - ord(c)))
            remain = (ord(lastchar) - ord(c)) - factor * factor
            direction = '-'

        result += '+' * factor + '[>' + direction * factor + '<-]>' + direction * remain + '.<'

        lastchar = c

    return result

File: rb-brainwords/brainwords.rb
def generateBFCode(text)

    bf = ""
    lastchar = 0

    text.each_byte { |c|

        factor = 0
        remain = 0
        direction = '+'

        if c > lastchar
            factor = (Math.sqrt(c - lastchar)).floor
            remain = (c - lastchar) - (factor * factor)
            direction = '+'
        else
            factor = (Math.sqrt(lastchar - c)).floor
            remain = (lastchar - c) - (factor * factor)
            direction = '-'
        end

        bf += ('+' * factor) + '[>' + (direction * factor) + '<-]>' + (direction * remain) + '.<'

        lastchar = c

    }

    return bf

end
